File: math_sequences/fibonacci_numbers.py
def fib_sequence(count: int) -> list:
    '''
    Return fibonacci sequence as list

    Parameters
    ----------
    count : int
        This is count of numbers in sequence
    '''
    if count < 1:
        return []
    elif count == 1:
        return [0]

    sequence = [0, 1]
    while len(sequence) < count:
        sequence.append(sequence[-2] + sequence[-1])

    return sequence


def fib_number(n: int) -> int:
    '''
    Return nth fibonacci number

    Parameters
    ----------
    n : int
        Index of number(Start at 1)
    '''
    if n <= 1:
        return 0

    a, b = 0, 1
    for _ in range(n-2):
        b, a = a + b, b

    return b


if __name__ == '__main__':
    print(fib_sequence(1))
    print(fib_sequence(-5))
    print(fib_sequence(3))
    print(fib_sequence(5))
    print(fib_sequence(10))

    print(fib_number(1))
    print(fib_number(-5))
    print(fib_number(3))
    print(fib_number(5))
    print(fib_number(10))

File: day_2020-05-06/fibonacci_numbers.c
#include <stdio.h>
#include <stdlib.h>

// Returns nth fibonacci number
unsigned long fib_number(unsigned int n){
    if(n <= 1)
        return 0;

    unsigned long a = 0, b = 1, tmp;
    for(int i = 0; i < (n - 2); i++){
        tmp = b;
        b = a + b;
        a = tmp;
    }

    return b;
}

// Returns fibonacci sequence
unsigned long* fib_sequence(size_t count){
    unsigned long* sequence;
    sequence = (unsigned long*)malloc(count * sizeof(unsigned long));

    if(count >= 1)
        sequence[0] = 0;
    if(count >= 2)
        sequence[1] = 1;

    for(int i = 2; i < count; i++){
        sequence[i] = sequence[i-1] + sequence[i-2];
    }

    return sequence;
}

int main(){
    // Test fib_number(0 to 9)
    for(int i = 0; i < 10; i++)
        printf("%d\n", fib_number(i));

    // Test fib_sequence(1 to 10)
    unsigned long *fi;
    for(int i = 0; i < 10; i++){
    	fi = fib_sequence(i);
    	for(int j = 0; j < i; j++)
    	    printf("%d ", fi[j]);
    	printf("\n");
    	free(fi);
    }

    return 0;
}
